File: Scripts/JavaScript_Node/mandelbrot.js
function mandelbrot(width, height, maxIter) {
    const buffer = new Int32Array(width * height);
    let checksum = 0n;

    const xMin = -2.0;
    const xMax = 1.0;
    const yMin = -1.5;
    const yMax = 1.5;

    const dx = (xMax - xMin) / (width - 1);
    const dy = (yMax - yMin) / (height - 1);

    let index = 0;
    for (let y = 0; y < height; y++) {
        const cy = yMin + y * dy;
        for (let x = 0; x < width; x++) {
            const cx = xMin + x * dx;

            let zx = 0.0;
            let zy = 0.0;
            let iter = 0;

            while (zx * zx + zy * zy <= 4.0 && iter < maxIter) {
                const zx2 = zx * zx - zy * zy + cx;
                zy = 2.0 * zx * zy + cy;
                zx = zx2;
                iter++;
            }

            buffer[index++] = iter;
            checksum += BigInt(iter);
        }
    }

    return checksum;
}

function main() {
    let width = 2000;
    let height = 2000;
    let maxIter = 1000;

    if (process.argv.length > 2) width = parseInt(process.argv[2], 10);
    if (process.argv.length > 3) height = parseInt(process.argv[3], 10);
    if (process.argv.length > 4) maxIter = parseInt(process.argv[4], 10);

    console.log(mandelbrot(width, height, maxIter).toString());
}

main();

File: Scripts/GNUOctave/mandelbrot.m
function mandelbrot_bench(varargin)
    width = 2000;
    height = 2000;
    maxIter = 1000;

    if nargin >= 1, width = str2double(varargin{1}); end
    if nargin >= 2, height = str2double(varargin{2}); end
    if nargin >= 3, maxIter = str2double(varargin{3}); end

    checksum = mandelbrot(width, height, maxIter);
    fprintf('%u\n', uint64(checksum));
end

function checksum = mandelbrot(width, height, maxIter)
    buffer = zeros(height, width, 'uint32');
    checksum = uint64(0);

    xMin = -2.0;
    xMax = 1.0;
    yMin = -1.5;
    yMax = 1.5;

    dx = (xMax - xMin) / (width - 1);
    dy = (yMax - yMin) / (height - 1);

    for y = 0:(height - 1)
        cy = yMin + y * dy;
        for x = 0:(width - 1)
            cx = xMin + x * dx;

            zx = 0.0;
            zy = 0.0;
            iter = uint32(0);

            while (zx * zx + zy * zy <= 4.0) && (iter < maxIter)
                zx2 = zx * zx - zy * zy + cx;
                zy = 2.0 * zx * zy + cy;
                zx = zx2;
                iter = iter + 1;
            end

            buffer(y + 1, x + 1) = iter;
            checksum = checksum + uint64(iter);
        end
    end
end

mandelbrot_bench()